editoast/editoast_derive/src/modelv2.rs

```diff
@@ -0,0 +1,648 @@
+        Ok(first.ident.clone())
+    }
+}
+
+impl darling::FromMeta for Identifier {
+    fn from_expr(expr: &syn::Expr) -> Result<Self> {
+        match expr {
+            syn::Expr::Path(path) => Ok(Identifier::Field(extract_ident_of_path(&path.path)?)),
+            syn::Expr::Tuple(tuple) => {
+                let mut idents = Vec::new();
+                for expr in tuple.elems.iter() {
+                    match expr {
+                        syn::Expr::Path(path) => {
+                            idents.push(extract_ident_of_path(&path.path)?);
+                        }
+                        _ => return Err(darling::Error::custom(
+                            "Model: invalid compound 'identifier' expression: must be a tuple of idents",
+                        )),
+                    }
+                }
+                Ok(Identifier::Compound(idents))
+            }
+            _ => Err(darling::Error::custom(
+                "Model: invalid 'identifier' expression",
+            )),
+        }
+    }
+}
+
+#[derive(FromDeriveInput, Debug)]
+#[darling(
+    attributes(model),
+    forward_attrs(allow, doc, cfg),
+    supports(struct_named)
+)]
+struct ModelOptions {
+    table: syn::Path,
+    #[darling(default)]
+    row: GeneratedType,
+    #[darling(default)]
+    changeset: GeneratedType,
+    #[darling(multiple, rename = "identifier")]
+    identifiers: Vec<Identifier>,
+    #[darling(default)]
+    preferred: Option<Identifier>,
+    data: ast::Data<util::Ignored, ModelFieldOption>,
+}
+
+#[derive(FromMeta, Default, Debug, PartialEq)]
+struct GeneratedType {
+    #[darling(default)]
+    type_name: Option<String>,
+    #[darling(default)]
+    derive: PathList,
+    #[darling(default)]
+    public: bool,
+}
+
+#[derive(FromField, Debug)]
+#[darling(attributes(model), forward_attrs(allow, doc, cfg))]
+struct ModelFieldOption {
+    ident: Option<syn::Ident>,
+    ty: syn::Type,
+    #[darling(default)]
+    builder_fn: Option<syn::Ident>,
+    #[darling(default)]
+    column: Option<String>,
+    #[darling(default)]
+    builder_skip: bool,
+    #[darling(default)]
+    identifier: bool,
+    #[darling(default)]
+    preferred: bool,
+    #[darling(default)]
+    primary: bool,
+    #[darling(default)]
+    json: bool,
+    #[darling(default)]
+    geo: bool,
+}
+
+#[derive(Debug, PartialEq)]
+struct ModelConfig {
+    model: syn::Ident,
+    table: syn::Path,
+    fields: Fields,
+    row: GeneratedType,
+    changeset: GeneratedType,
+    identifiers: HashSet<Identifier>, // identifiers ⊆ fields
+    preferred_identifier: Identifier, // preferred_identifier ∈ identifiers
+    primary_field: Identifier,        // primary_field ∈ identifiers
+}
+
+#[derive(Debug, PartialEq, Clone)]
+struct ModelField {
+    ident: syn::Ident,
+    column: String,
+    builder_ident: syn::Ident,
+    ty: syn::Type,
+    builder_skip: bool,
+    identifier: bool,
+    preferred: bool,
+    primary: bool,
+    json: bool,
+    geo: bool,
```

Comment:
We should have an enum here. When parsing the `ModelFieldOption` if multiple option is set then trigger an error.